src/models/base.es6.js

```diff
@@ -1,59 +1,41 @@
-import { EventEmitter } from 'events';
+const THING_ID_REGEX = new RegExp('t\\d_[0-9a-z]+', 'i');
 
 class Base {
-  constructor (props={}, opts={}) {
+  constructor (props={}) {
     this.props = {};
-    this.validators = opts.validators || [];
-    this.formatters = opts.formatters || [];
-    this.emitter = opts.emitter || new EventEmitter();
 
-    for (var p in props) {
+    for (let p in props) {
       this.props[p] = props[p];
     }
   }
 
+  get _type () {
+    return this.constructor.name;
+  }
+
   get (name) {
     return this.format(name, this.props[name]);
   }
 
-  set (name, value, emit = { value: true, base: true}) {
+  set (name, value) {
     if (typeof name === 'object') {
-      for (var n in name) {
-        this.set(n, name[n], { value: true });
-      }
-
-      this.emit('set', name);
+      Object.assign(this.props, name);
     } else {
       this.props[name] = value;
-
-      if(emit && emit.base) {
-        this.emit('set', name, value);
-      }
-
-      if(emit && emit.value) {
-        this.emit('set:' + name, value);
-      }
     }
   }
 
-  emit (...args) {
-    this.emitter.emit.apply(this, args);
-  }
-
-  on (...args) {
-    this.emitter.on.apply(this, args);
-  }
-
   validate () {
-    if (!this.validators) {
+    const validators = this.validators();
```

Comment:
How would you feel about a base implementation of `validators`? it looks like it's undefined in this instance.